core-api/core_api/routes/chat.py

```diff
@@ -39,62 +37,38 @@
 )
 
 
-async def route_chat(
-    chat_request: ChatRequest, user_uuid: UUID, routable_chains: dict[str, Tool]
-) -> tuple[Runnable, ChainInput]:
-    question = chat_request.message_history[-1].text
-
-    selected_chain = None
-
-    def select_chat_chain(chat_request: ChatRequest, routable_chains: dict[str, Runnable]) -> Runnable:
-        if chat_request.selected_files:
-            return routable_chains.get("chat/documents")
-        else:
-            return routable_chains.get("chat")
-
-    # Match keyword
-    route_match = re_keyword_pattern.search(question)
-    route_name = route_match.group()[1:] if route_match else None
-    selected_chain, description = routable_chains.get(route_name, select_chat_chain(chat_request, routable_chains))
-
-    params = ChainInput(
-        question=chat_request.message_history[-1].text,
-        file_uuids=[str(f.uuid) for f in chat_request.selected_files],
-        user_uuid=str(user_uuid),
-        chat_history=[
-            ChainChatMessage(role=message.role, text=message.text) for message in chat_request.message_history[:-1]
-        ],
-    )
-
-    log.info("Routed to %s", route_name)
-    log.info("Selected files: %s", chat_request.selected_files)
-
-    return selected_chain, params
-
-
 @chat_app.post("/rag", tags=["chat"])
 async def rag_chat(
     chat_request: ChatRequest,
     user_uuid: Annotated[UUID, Depends(get_user_uuid)],
-    routable_chains: Annotated[dict[str, Tool], Depends(get_routable_chains)],
+    redbox: Annotated[Redbox, Depends(get_redbox)],
 ) -> ChatResponse:
     """REST endpoint. Get a LLM response to a question history and file."""
-    selected_chain, params = await route_chat(chat_request, user_uuid, routable_chains)
-    return (selected_chain | map_to_chat_response).invoke(params.dict())
+    state = ChainState(
+        query=ChainInput(
+            question=chat_request.message_history[-1].text,
+            file_uuids=[f.uuid for f in chat_request.selected_files],
+            user_uuid=user_uuid,
+            chat_history=[
+                ChainChatMessage(role=message.role, text=message.text) for message in chat_request.message_history[:-1]
+            ],
+        ),
+    )
+    return await (redbox.graph | map_to_chat_response).ainvoke(state)
 
 
 @chat_app.get("/tools", tags=["chat"])
 async def available_tools(
-    routable_chains: Annotated[dict[str, tuple[Runnable, str]], Depends(get_routable_chains)],
+    redbox: Annotated[Redbox, Depends(get_redbox)],
```

Comment:
are we using this in the front end? if not i would remove it as its more surface are to test/maintain/get-out-of-date

Reply:
we aren't but we should, it's in as a first step in some features I'd like to work on in the frontend